src/stream/stream/mod.rs

```diff
@@ -921,6 +924,52 @@ extension_trait! {
             Skip::new(self, n)
         }
 
+        #[doc = r#"
+            Applies a falliable function to each element in a stream, stopping at first error and returning it.
+
+            # Examples
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use std::collections::VecDeque;
+            use std::sync::mpsc::channel;
+            use async_std::prelude::*;
+
+            let (tx, rx) = channel();
+
+            let s: VecDeque<usize> = vec![1, 2, 3].into_iter().collect();
+            let s = s.try_for_each(|v| {
```

Comment:
Do you think it's possible to make this operate on `|v| async {`?

Reply:
I will update #262 as soon as this is merged, yes.